src/ldp/http/RawBodyParser.ts

```diff
@@ -39,35 +40,28 @@ export class RawBodyParser extends BodyParser {
   private parseMetadata(input: HttpRequest): RepresentationMetadata {
     const contentType = /^[^;]*/u.exec(input.headers['content-type']!)![0];
 
-    const metadata: RepresentationMetadata = {
-      raw: [],
-      contentType,
-    };
+    const metadata: RepresentationMetadata = new RepresentationMetadata().add(CONTENT_TYPE, contentType);
 
     const { link, slug } = input.headers;
 
     if (slug) {
       if (Array.isArray(slug)) {
         throw new UnsupportedHttpError('At most 1 slug header is allowed.');
       }
-      metadata.slug = slug;
+      metadata.set(SLUG, slug);
     }
 
     // There are similarities here to Accept header parsing so that library should become more generic probably
     if (link) {
-      metadata.linkRel = {};
       const linkArray = Array.isArray(link) ? link : [ link ];
       const parsedLinks = linkArray.map((entry): { url: string; rel: string } => {
         const [ , url, rest ] = /^<([^>]*)>(.*)$/u.exec(entry) ?? [];
         const [ , rel ] = /^ *; *rel="(.*)"$/u.exec(rest) ?? [];
         return { url, rel };
       });
       parsedLinks.forEach((entry): void => {
-        if (entry.rel) {
-          if (!metadata.linkRel![entry.rel]) {
-            metadata.linkRel![entry.rel] = new Set();
-          }
-          metadata.linkRel![entry.rel].add(entry.url);
+        if (entry.rel === 'type') {
+          metadata.set(RDF_TYPE, entry.url);
```

Comment:
Minor nit, but we're looping over everything here, as opposed to getting the first (if any) and then stopping.